linera-rpc/src/simple/client.rs

```diff
@@ -149,24 +149,18 @@ impl ValidatorNode for SimpleClient {
         &self,
         hash: CryptoHash,
     ) -> Result<ConfirmedBlockCertificate, NodeError> {
-        self.query::<Certificate>(RpcMessage::DownloadCertificate(Box::new(hash)))
-            .await?
-            .try_into()
-            .map_err(|_| NodeError::UnexpectedCertificateValue)
+        self.query::<ConfirmedBlockCertificate>(RpcMessage::DownloadCertificate(Box::new(hash)))
+            .await
     }
 
     async fn download_certificates(
         &self,
         hashes: Vec<CryptoHash>,
     ) -> Result<Vec<ConfirmedBlockCertificate>, NodeError> {
-        self.query::<Vec<Certificate>>(RpcMessage::DownloadCertificates(Box::new(hashes)))
-            .await?
-            .into_iter()
-            .map(|cert| {
-                cert.try_into()
-                    .map_err(|_| NodeError::UnexpectedCertificateValue)
-            })
-            .collect()
+        self.query::<Vec<ConfirmedBlockCertificate>>(RpcMessage::DownloadCertificates(Box::new(
```

Comment:
(Also not your fault, but: `Box`ing a `Vec` is almost redundant, I think.)

Reply:
Removed in [Don't Box Vectors,](https://github.com/linera-io/linera-protocol/pull/2976/commits/78edcaeb33fee655e9ba3a9f8e1a6d049028975d)